src/test/scala/org/kibanaLoadTest/simulation/branch/GenericJourney.scala

```diff
@@ -0,0 +1,195 @@
+      journey: Journey,
+      config: KibanaConfiguration
+  ): ChainBuilder = {
+    var steps: ChainBuilder = exec()
+    var priorRequest: Option[org.kibanaLoadTest.simulation.branch.Request] =
+      Option.empty
+    for (trace <- journey.requests) {
+      val request = trace.request
+      val priorTimestamp =
+        priorRequest.map(_.timestamp).getOrElse(request.timestamp)
+      val pauseDuration =
+        Math.max(0L, request.timestamp.getTime - priorTimestamp.getTime)
+      if (pauseDuration > 0L) {
+        steps = steps.pause(pauseDuration.toString, TimeUnit.MILLISECONDS)
+      }
+
+      steps = steps.exec(ApiCall.execute(request, config))
+      priorRequest = Option(request)
+    }
+    steps
+  }
+
+  def scenarioForStage(
+      steps: ChainBuilder,
+      scenarioName: String
+  ): ScenarioBuilder = {
+    scenario(scenarioName).exec(steps)
+  }
+
+  def populationForStage(
+      scn: ScenarioBuilder,
+      stage: List[Step]
+  ): PopulationBuilder = {
+    val steps = ListBuffer[ClosedInjectionStep]()
+    for (step <- stage) {
+      val injectionStep = step.action match {
+        case "constantConcurrentUsers" =>
+          constantConcurrentUsers(step.maxUsersCount) during (getDuration(
+            step.duration
+          ))
+        case "rampConcurrentUsers" =>
+          rampConcurrentUsers(
+            step.minUsersCount.get
+          ) to step.maxUsersCount during (getDuration(step.duration))
+        case _ =>
+          throw new IllegalArgumentException(s"Invalid action: ${step.action}")
+      }
+      steps.addOne(injectionStep)
+    }
+    scn.inject(steps)
+  }
+
+  private val journeyPath = Option(System.getProperty("journeyPath")) match {
+    case Some(v) => new File(v)
+    case _ =>
+      throw new IllegalArgumentException(
+        "The journeyPath system property is mandatory but no value is provided"
+      )
+  }
+  if (!journeyPath.isFile || !journeyPath.getName.endsWith(".json")) {
+    throw new IllegalArgumentException(
+      s"Provide path to valid json journey file using journeyPath system var, found '$journeyPath'"
+    )
+  }
+
+  private val journeyFile = Using(fromFile(journeyPath)) { f =>
+    f.getLines().mkString("\n")
+  }
+  private val journey = journeyFile.get.parseJson.convertTo[Journey]
+
+  // These values are valid as long we use FTR to start Kibana server
+  private val kibanaHost = "http://localhost:5620"
+  private val esHost = "http://localhost:9220"
+  private val providerType = "basic"
+  private val providerName = "basic"
+  private val username = "elastic"
+  private val password = "changeme"
+
+  val appConfig: KibanaConfiguration = new KibanaConfiguration(
+    kibanaHost,
+    journey.kibanaVersion,
+    esHost,
+    username,
+    password,
+    providerType,
+    providerName
+  )
+
+  val httpHelper = new HttpHelper(appConfig)
+  var httpProtocol: HttpProtocolBuilder =
+    httpHelper.getProtocol.baseUrl(appConfig.baseUrl).disableFollowRedirect
```

Comment:
By default, Gatling automatically follow redirects in case of 301, 302, 303, 307 or 308 response status code. And we requests with 302 status code in the list, so I'm disabling behaviour and only run what is in APM traces
https://gatling.io/docs/gatling/reference/current/http/protocol/#disablefollowredirect